contracts/Rewards/RewardsDistributor.sol

```diff
@@ -265,6 +237,39 @@ contract RewardsDistributor is ExponentialNoError, Ownable2StepUpgradeable, Acce
         }
     }
 
+    /**
+     * @notice Set REWARD TOKEN last rewarding block timestamp for the specified markets, used when contract is time based
+     * @param vTokens The markets whose REWARD TOKEN last rewarding block to update
+     * @param supplyLastRewardingBlockTimestamps New supply-side REWARD TOKEN last rewarding block timestamp for the corresponding market
+     * @param borrowLastRewardingBlockTimestamps New borrow-side REWARD TOKEN last rewarding block timestamp for the corresponding market
+     */
+    function setLastRewardingBlockTimestamps(
```

Comment:
Note: we'll need to grant permission for this in the ACM, to the Normal timelock